force-app/main/default/classes/DMLOperations.cls

```diff
@@ -31,13 +31,18 @@ public with sharing class DMLOperations {
      */
     public static Id insertNewAccount() {
         // Create a new Account instance
-
-        // Set some fields on the Account
-
-        // Insert the Account into the database
-
-        // Return the Id of the newly created Account
-        return null;
+        Account acc = new Account(Name = 'Test Account');
+        // Check CRUD prior to DML operation
+        if (Account.SObjectType.getDescribe().isCreateable() &&
+            Schema.SObjectType.Account.fields.Name.isCreateable()){
+                // Insert the new account into the database if the User has access
+                insert acc;
+            } else {
+                // return null if the User does not have access and do not insert the account
+                return null;
+            }
+        // Return the Id of the newly created Account if the User has access to create the account
+        return acc.Id;
```

Comment:
You went above and beyond here. Good work. I would recommend that you try with the new User Mode and System Mode.  In a real production environment I  probably would not return null. I would throw an exception